controllers/productController.js

```diff
@@ -1,13 +1,29 @@
 const productService = require('../services/productService');
-const { catchError } = require('../middlewares/error');
+const { catchError, baseError } = require('../middlewares/error');
 
 const getProductById = catchError(async (req, res) => {
   const { productId } = req.params;
-  const { mainCategory } = req.body;
-  const product = await productService.getProductById(productId, mainCategory);
+  if (!productId) {
+    return new baseError('NOT_PRODUCTID', 400);
+  }
```

Comment:
1. if 시작 전 공백 추가해 주세요.
2. 데이터가 없을 때는 404 상태코드를 사용합니다.
3. 메시지가 명확한 것 같지 않아 아래와 같이 수정해 봤습니다.
```suggestion
  
  if (!productId) {
    return new baseError('PRODUCT_DOES_NOT_EXIST', 404);
  }
```